[IOI] Wootube 1주차 기능 구현

MrKwon opened the pull request:
User
* [백엔드] 회원 생성/조회/수정/삭제
* [백엔드] 회원 관련 인터셉터 (로그인 한 상황에서 회원가입 페이지, 로그인 페이지 이동 불가, 로그인 하지 않은 상황에서 마이페이지 이동 불가)
* [프론트엔드] 회원 가입, 로그인 페이지, 마이페이지

Video
* [백엔드] 영상 생성/조회/수정/삭제
* [프론트엔드] 영상 생성/조회/수정

Comment
* [백엔드] 댓글  생성/조회/수정/삭제
* [프론트엔드] 댓글 생성/조회/수정/삭제
* [백엔드] 대댓글 생성/조회/수정/삭제

MrKwon committed:
init: 프로젝트 초기 설정

MrKwon committed:
[루피] User 도메인 (#18)

* docs: README.md User 요구사항 추가

* chore: application.properties 설정

* chore: passay 의존성 추가

* Password Validation 을 쉽게 해주는 라이브러리

* feat: Password 어노테이션 구현

* Password interface 구현
* PasswordConstraintValidator 클래스 구현

* feat: User 도메인

* test: User 테스트

backlo committed:
[포도당] 영상 조회 프론트(#20)

* chore: build.gradle 초기화

* refactor: index.html 페이지 동적 움직이겠금 수정 및 중복 코드 삭제 및 분할

* chore: build.gradle devtool 추가

* feat: video.html 구현

* feat: livereload 설정

* feat: video.html 코드 수정

backlo committed:
[티버, 포도당] 영상 CRUD (#25)

* chore: build.gradle 초기화

* refactor: index.html 페이지 동적 움직이겠금 수정 및 중복 코드 삭제 및 분할

* chore: build.gradle devtool 추가

* feat: video.html 구현

* feat: livereload 설정

* feat: video.html 코드 수정

* refactor: index페이지로 이동하는 HomeController 생성 및 TestController 삭제

* chore: jupiter, webflux, aws, modelmapper dependency 추가

* feat: 동영상 등록 기능 구현

* chore: aws의 s3 및 파일 업로드 용량 관련 설정 추가

* feat: Video 조회 기능 구현

* feat: Video 수정 기능 구현

* feat: Video 삭제 기능 구현

* refactor: video update 메소드 수정

* refactor: video 생성 시 해당 페이지로 이동하도록 수정

* refactor: 패키지 구조 변경

* chore: git 리버트

티버 포도당 깃 합치기

* refactor: package 구조 수정

* test: test 구조 수정

* feat: video-edit.html 중복 제거

* chore: properties, git ignore 설정

* chore: build.gradle s3Mock dependency 추가

* refactor: package 구조 변경

* feat: 비디오 프론트 코드 동적으로 수정

* test: video 등록 test 작성

* feat: 비디오 삭제 기능 구현

* test: VideoController test code 작성

* refactor: config 클래스명 변경

* refactor: Video 객체의 setContentPath메소드를 사용하는 부분 수정

* fix: 충돌 시 해결하지 않은 부분 수정

* refactor: VideoResponseDto 생성 및 VideoRequestDto 필드 수정

* refactor: Video 클래스 패키지 이동

* style: 코드 컨벤션을 위한 수정

* test: 중복 코드 추출

* refactor: Video service 리펙토링

* refactor: Video controller 패키지 리펙토링 및 @AutoConfigureWebTestClient 추가

* test: videos service test 구현

* feat: videos service delete 메소드 deleteById로 변경

* refactor: 코드 리펙토링

* test: test 코드 리펙토링

MrKwon committed:
[루피] 회원 CRUD (#26)

* feat: signup.html 과 login.html 작성

* feat: form을 보여주는 컨트롤러 메서드 작성

* signup form
* login form

* chore: WebTestClient 를 위한 Dependency 추가

* test: UserController 회원가입 로그인 폼 메서드

* feat: 회원을 등록하는 기능 추가

* SignUpRequestDto 회원 등록시 필요한 정보를 받아오기 위한 Dto

* test: 회원 추가 기능

* UserServiceTest
* UserControllerTest

* chore: static path 경로 지정 삭제

* fix: 리소스 파일 경로 수정

* feat: IndexController 메인화면 보여주는 기능

* feat: findByEmail 추가

* style: SignUpRequestDto 메서드 이름 변경

* test: findByEmail 메서드

* feat: User 패스워드 확인 메서드 구현

* NotMatchPasswordException 추가

* test: matchPassword 메서드 테스트 추가

* feat: 로그인 기능 구현

* LogInRequestDto 추가
* UserController login 메서드
* UserService login 메서드
* NotFoundUserException 예외 클래스

* test: UserService 로그인 성공, 실패 테스트 추가

* test: UserController 유저 로그인 성공, 실패 테스트 추가

* feat: html 과 Controller 연결

* feat: header 로그인 처리 로직 추가

* session에 user 가 있으면 프로필 사진이 나오게 함
* session에 user 가 없으면 로그인 버튼이 나오게 함

* feat: 로그인 세션 처리 기능

* refactor: 패키지 구조 수정

* feat: 로그아웃 기능 추가

* header.html 로그아웃 기능 추가

* feat: 마이페이지 추가

* mypage.html
* UserController 마이페이지 메서드

* feat: header 마이페이지, 로그인 버튼 추가

* feat: sidebar.html 홈 href 수정

* feat: mypage.html 구현

* feat: User 정보 수정 기능 추가

* EditUserRequestDto : 유저가 입력한 수정 정보를 받아오기 위한 Dto
* User : updateName 메서드 추가
* UserService : update 메서드 추가
* UserController : edit 추가

* feat: mypage.html 회원 탈퇴 버튼 추가

* feat: User 삭제 기능 추가

* UserService delete 메서드 추가, 중복 제거
* UserController deleteUser 메서드 추가, edit 메서드 이름 변경

* test: User 도메인 테스트 접근 제어자 수정

* feat: 아이디 중복 체크 Rest Controller 메서드 추가

* UserRestController
* UserService 중복을 체크하는 checkDuplicate 메서드 추가
* EmailCheckRequestDto 와 EmailCheckRequestDto 추가

* feat: signup.html 중복체크 로직 추가

* chore: Rest Assured 의존성 추가

* feat: EmailCheckResponseDto equals, hashcode 롬복 어노테이션 추가

* test: UserService 아이디 중복 확인 테스트

* test: UserRestController 중복 확인 테스트 추가

* feat: RedirectionArgumentResolver 구현

* 요청을 보낸 주소로 다시 쉽게 Redirect 하기 위해 도와줌

* feat: LoginFailedException 추가

* feat: UserService 로그인 메서드 예외 처리 로직 추가

* 이메일, 비밀번호 둘 중 하나가 틀려도 동일한 메시지를 전달하도록 로직 수정

* test: LoginFailedExceptionAdvice 추가

* feat: login.html 에러를 보여주는 div 태그 추가

* fix: UserService login 예외 처리 로직 변경 관련된 테스트 수정

* feat: 로그인 했을 때를 처리하기 위한 인터셉터

* LoginInterceptor
* LoginInterceptorConfig Config 에 추가

* feat: 로그인 안 했을 때를 처리하기 위한 인터셉터

* NotLoginInterceptor
* NotLoginInterceptorConfig Config 에 추가

* style: image 변경

* test: User updateName 메서드 테스트 추가

* test: IndexController 테스트 추가

* style: UserService 메서드 이름 변경

* refactor: UserRestService 분리

* UserRestController 에서 사용하는 UserService 의 메서드를 UserRestService 로 분리

* test: UserRestService 테스트 작성

* style: signup.html 중복제거 및 코드 분리

* style: login.html 중복제거 및 코드 분리

* refactor: 로직 단순화

* style: 클래스 이름 변경

* SessionUser 에서 UserSession 으로 변경
* LoginUserSessionManager 에서 UserSessionManager 로 변경

* style: 상수로 분리

* style: default 생성자 접근 제어자 변경

* EmailCheckRequestDto
* EmailCheckResponseDto

* style: 불필요한 Setter 어노테이션 제거 및 디폴트 생성자 추가

* test: 로그인하지 않고 마이 페이지 요청 시도 테스트 추가

* refactor: UserSession of 메서드 제거 후 로직 수정

* feat: UserAssembler 추가

* SignUpRequestDto 의 toUser 메서드 제거

hyojaekim committed:
[효오, 비모] 댓글 CRUD (#27)

* chore: dependency 추가

* feat: Comment 작성 기능 추가

Comment 작성 API 추가

* feat: Comment 수정 기능 추가

Comment 수정 API 추가

* feat: Comment 삭제 기능 추가

Comment 삭제 API 추가

* chore: 불필요한 설정 제거

* feat: Comment 작성 기능 추가

1. 영상 페이지에 댓글 HTML 합치기
2. 댓글 작성 관련 Javascript 코드 추가
3. 댓글 작성 관련 CSS 추가

* refactor: Comment 예외 처리 가이드 추가

* feat: 없는 Comment에 대한 Exception 추가

* refactor: Comment 생성자 Lombok 적용 및 Column 속성 추가

* refactor: Comment 연관 관계 가이드 추가

* test: 댓글 생성 테스트 추가

* test: Comment 수정 테스트 추가

* test: 댓글 삭제 테스트 추가

* refactor: Comment Test 가이드 추가

* test: 존재하지 않는 comment update 테스트 추가

* test: 존재하지 않는 Comment 삭제 테스트 추가

* style: 컨벤션에 맞춰 코드 포맷 수정

* test: Comment Service 댓글 생성 테스트 추가

* refactor: Comment EqualsAndHashCode 추가

* test: Comment Service 댓글 수정 테스트 추가

* refactor: CommentResponse 생성자 제거

* refactor: 패키지 이동

* test: Comment Service 댓글 삭제 테스트 추가

1. 댓글이 없는 경우
2. 댓글이 있는 경우

* test: Comment Service 존재하지 않은 댓글 수정 테스트 추가

* refactor: Comment Service 예외 처리 가이드 추가

* refactor: 구독 버튼 테두리 제거

* style: 패키지 이동

* style: Comment 관련 Dto 이름 수정

* style: Comment 조회 기능 가이드 추가

* feat: comment front 코드 추가

1. 수정
2. 삭제

* feat: Reply Entity 생성

1. 업데이트 로직 추가

* feat: Reply 생성 및 수정 기능 추가

* feat: Reply 예외 처리

1. 존재하지 않는 답글인 경우
2. ExceptionHandler 추가

* test: Reply 생성 및 수정 테스트 추가

- 상속을 이용한 테스트 중복 제거

* refactor: Comment 찾는 메소드 추가

* feat: comment front 코드 추가

프로필 사진 및 버튼 이벤트 추가

* refactor: 공통 코드 분리

* refactor: Reply 메서드 분리

- 댓글이랑 매치되지 않는 경우 예외를 던지는 로직

* feat: Reply 삭제 기능 추가

- 중복 Path 수정
- 예외 처리

* test: Reply 삭제 테스트 추가

1. 답글을 삭제하는데 성공하는 테스트
2. 다른 댓글의 답글을 삭제하는 경우 실패하는 테스트
3. 존재하지 않는 답글을 삭제하는 경우 실패하는 테스트

* test: Rest Assured 적용

* fix: 테스트 어노테이션 추가

wmakerjun commented on src/main/resources/templates/mypage.html:
```diff
@@ -0,0 +1,155 @@
+                                            취소
+                                        </v-btn>
+                                        <v-btn color="blue darken-1"
+                                               onclick="editSubmitBtnHandler()" text>저장
```
Vue를 이용해서 핸들링하고 있으니까 onclick 대신에 44라인처럼 @click로 통일해서 쓰면 어떨까요?

MrKwon replied:
피드백 반영하였습니다 !

wmakerjun commented on src/main/resources/templates/mypage.html:
```diff
@@ -0,0 +1,155 @@
+        }
+    }
+
+    const deleteSubmitBtnHandler = () => {
```
Vue로 생성한 컴포넌트 안에서 사용되는 메서드이기 때문에 지금 처럼 전역변수에 할당하여 사용하기 보다는, Vue의 methods를 이용하여 정리하면 좋을 것 같아요 :) 전역변수로 무언가를 선언하는 것은 나중에 외부 라이브러리를 쓰면서 충돌이 날 가능성이 있기 때문에 주의해서 사용하면 좋답니다. 😉

MrKwon replied:
mypage의 [다른 피드백](https://github.com/woowacourse/miniprojects-2019/pull/11#discussion_r314943748) 을 반영하면서 반영하였습니다 👍

wmakerjun commented on src/main/resources/templates/fragments/signup.html:
```diff
@@ -0,0 +1,164 @@
+            'Content-Type': 'application/json'
+        }
+
+        const _success = 'green'
```
자바스크립트에서 변수 앞에 _를 붙이는 것은 유효범위를 제한시키기 위한 private 상태인 변수를 네이밍할 때 사용하는 컨벤션입니다. 현재는 전역변수로 만든 상태에서 앞에 _를 붙이는 것은 오해의 여지가 있어 보입니다. [링크](https://itmining.tistory.com/72)를 참조해주세요 :)

MrKwon replied:
넵 ! 감사합니다 ! 피드백 반영하였습니다 !

wmakerjun commented on src/main/resources/static/js/reply.js:
```diff
@@ -0,0 +1,58 @@
+        }
+
+        function toggleReplyWrite(event) {
+            if (event.target.className === "reply-toggle-btn") {
```
className 메서드보다는 element.classList.contains( 'someclass' ); 와 같이 contains 메서드를 이용하는게 더 좋아보입니다. className은 return type이 string이기 때문에 다른사람과 협업과정에서 해당 엘리먼트에 class를 추가로 작성하거나 순서가 바뀌는 경우 원치 않은 오류가 발생할 가능성이 있어보여요~ 아래 링크에서 참조해주세요 :)

[className](https://www.w3schools.com/jsref/prop_html_classname.asp)
[contains](https://www.w3schools.com/jsref/prop_element_classlist.asp)

phs1116 commented on src/main/java/com/wootube/ioi/WootubeApplication.java:
```diff
@@ -0,0 +1,19 @@
+
+@SpringBootApplication
+public class WootubeApplication {
+    public static final String APPLICATION_LOCATIONS = "spring.config.location="
```
설정 파일 위치를 다시 설정할 필요가 있을까요?
내부적으로 프로파일에 따라 클래스패스의 application.yml 혹은 application.propoerties를 불러올 뿐더러
aws 설정파일이 별도로 필요하다면 설정파일에서 include를 통해서 불러올수 있습니다.

yjll1019 replied:
네 수정했습니다!

phs1116 commented on src/main/java/com/wootube/ioi/domain/model/Comment.java:
```diff
@@ -0,0 +1,51 @@
+    private String contents;
+
+    @CreationTimestamp
+    private LocalDateTime createTime;
```
createTime, updateTime 등 공통적으로 사용되는 부분을 추상화하여 사용해보면 어떨까요?

hyojaekim replied:
Id, CreateTime, UpdateTime가 중복되고 있어서 BaseEntity로 추상화하였습니다.

phs1116 replied:
이름 좋네요 👍

phs1116 commented on src/main/java/com/wootube/ioi/domain/model/Comment.java:
```diff
@@ -0,0 +1,51 @@
+    @UpdateTimestamp
+    private LocalDateTime updateTime;
+
+//    @ManyToOne
```
사용 안하는 부분은 제거해주세요 :)

phs1116 commented on src/main/java/com/wootube/ioi/domain/model/Reply.java:
```diff
@@ -0,0 +1,50 @@
+
+@Getter
+@Entity
+@NoArgsConstructor
```
JPA를 위해 사용하는 기본생성자라면 private 생성자로 선언하여 의도치않은 생성을 막을 수 있습니다. :)

phs1116 commented on src/main/java/com/wootube/ioi/domain/model/User.java:
```diff
@@ -0,0 +1,62 @@
+
+@Entity
+@Getter
+@EqualsAndHashCode(exclude = {"name", "email", "password"})
```
엔티티의 유일 식별값은 엔티티의 ID입니다.

MrKwon replied:
피드백 반영하였습니다. 감사합니다. !

phs1116 commented on src/main/java/com/wootube/ioi/domain/model/Video.java:
```diff
@@ -0,0 +1,65 @@
+@Setter
+@NoArgsConstructor
+@DynamicUpdate
+public class Video {
```
다른 엔티티와 다르게 EqualsAndHashCode가 빠진것 같네요.
Setter도 특별한 사유가 없는 이상 사용 안하도록 하는게 좋을 것 같습니다. :)

yjll1019 replied:
네~ 수정했습니다!

phs1116 commented on src/main/java/com/wootube/ioi/service/CommentService.java:
```diff
@@ -0,0 +1,59 @@
+                comment.getUpdateTime());
+    }
+
+//    public List<CommentResponseDto> findByVideoId(Long videoId) {
```
사용 안하는 코드에 대한 주석은 모두 제거해주세요!

phs1116 commented on src/main/java/com/wootube/ioi/service/UserService.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    public User createUser(SignUpRequestDto signUpRequestDto) {
+        return userRepository.save(UserAssembler.toDomain(signUpRequestDto));
```
현재 정도의 활용에서는 UserAssembler가 굳이 필요할까? 라는 생각이 드네요. :) 
또한 다른 서비스들은 대부분 생성자를 통해 생성하는데 코드가 통일성이 없는 것 같습니다.

MrKwon replied:
ModelMapper 를 사용하는 것이 별로인 것 같아서 Assembler 를 두고 사용했는데, 팀 내에서 ModelMapper 를 사용하는 쪽으로 결론이 나서 ModelMapper 를 사용하는 로직으로 변경하였습니다 !

MrKwon committed:
[IOI] 의존성 수정 (#39)

* feat: signup.html 과 login.html 작성

* feat: form을 보여주는 컨트롤러 메서드 작성

* signup form
* login form

* chore: WebTestClient 를 위한 Dependency 추가

* test: UserController 회원가입 로그인 폼 메서드

* feat: 회원을 등록하는 기능 추가

* SignUpRequestDto 회원 등록시 필요한 정보를 받아오기 위한 Dto

* test: 회원 추가 기능

* UserServiceTest
* UserControllerTest

* chore: static path 경로 지정 삭제

* fix: 리소스 파일 경로 수정

* feat: IndexController 메인화면 보여주는 기능

* feat: findByEmail 추가

* style: SignUpRequestDto 메서드 이름 변경

* test: findByEmail 메서드

* feat: User 패스워드 확인 메서드 구현

* NotMatchPasswordException 추가

* test: matchPassword 메서드 테스트 추가

* feat: 로그인 기능 구현

* LogInRequestDto 추가
* UserController login 메서드
* UserService login 메서드
* NotFoundUserException 예외 클래스

* test: UserService 로그인 성공, 실패 테스트 추가

* test: UserController 유저 로그인 성공, 실패 테스트 추가

* feat: html 과 Controller 연결

* feat: header 로그인 처리 로직 추가

* session에 user 가 있으면 프로필 사진이 나오게 함
* session에 user 가 없으면 로그인 버튼이 나오게 함

* feat: 로그인 세션 처리 기능

* refactor: 패키지 구조 수정

* feat: 로그아웃 기능 추가

* header.html 로그아웃 기능 추가

* feat: 마이페이지 추가

* mypage.html
* UserController 마이페이지 메서드

* feat: header 마이페이지, 로그인 버튼 추가

* feat: sidebar.html 홈 href 수정

* feat: mypage.html 구현

* feat: User 정보 수정 기능 추가

* EditUserRequestDto : 유저가 입력한 수정 정보를 받아오기 위한 Dto
* User : updateName 메서드 추가
* UserService : update 메서드 추가
* UserController : edit 추가

* feat: mypage.html 회원 탈퇴 버튼 추가

* feat: User 삭제 기능 추가

* UserService delete 메서드 추가, 중복 제거
* UserController deleteUser 메서드 추가, edit 메서드 이름 변경

* test: User 도메인 테스트 접근 제어자 수정

* feat: 아이디 중복 체크 Rest Controller 메서드 추가

* UserRestController
* UserService 중복을 체크하는 checkDuplicate 메서드 추가
* EmailCheckRequestDto 와 EmailCheckRequestDto 추가

* feat: signup.html 중복체크 로직 추가

* chore: Rest Assured 의존성 추가

* feat: EmailCheckResponseDto equals, hashcode 롬복 어노테이션 추가

* test: UserService 아이디 중복 확인 테스트

* test: UserRestController 중복 확인 테스트 추가

* feat: RedirectionArgumentResolver 구현

* 요청을 보낸 주소로 다시 쉽게 Redirect 하기 위해 도와줌

* feat: LoginFailedException 추가

* feat: UserService 로그인 메서드 예외 처리 로직 추가

* 이메일, 비밀번호 둘 중 하나가 틀려도 동일한 메시지를 전달하도록 로직 수정

* test: LoginFailedExceptionAdvice 추가

* feat: login.html 에러를 보여주는 div 태그 추가

* fix: UserService login 예외 처리 로직 변경 관련된 테스트 수정

* feat: 로그인 했을 때를 처리하기 위한 인터셉터

* LoginInterceptor
* LoginInterceptorConfig Config 에 추가

* feat: 로그인 안 했을 때를 처리하기 위한 인터셉터

* NotLoginInterceptor
* NotLoginInterceptorConfig Config 에 추가

* style: image 변경

* test: User updateName 메서드 테스트 추가

* test: IndexController 테스트 추가

* style: UserService 메서드 이름 변경

* refactor: UserRestService 분리

* UserRestController 에서 사용하는 UserService 의 메서드를 UserRestService 로 분리

* test: UserRestService 테스트 작성

* style: signup.html 중복제거 및 코드 분리

* style: login.html 중복제거 및 코드 분리

* refactor: 로직 단순화

* style: 클래스 이름 변경

* SessionUser 에서 UserSession 으로 변경
* LoginUserSessionManager 에서 UserSessionManager 로 변경

* style: 상수로 분리

* style: default 생성자 접근 제어자 변경

* EmailCheckRequestDto
* EmailCheckResponseDto

* style: 불필요한 Setter 어노테이션 제거 및 디폴트 생성자 추가

* test: 로그인하지 않고 마이 페이지 요청 시도 테스트 추가

* refactor: UserSession of 메서드 제거 후 로직 수정

* feat: UserAssembler 추가

* SignUpRequestDto 의 toUser 메서드 제거

* chore: spring cloud 관련 dependency 수정

* test: 기존의 테스트 코드로 롤백

* @ExtendWith 어노테이션 삭제

* test: 클래스 이름 변경

* style: 메서드 이름 변경

* fix: aws.yml 관련 설정 제거

* feat: S3 연동 재구현

* AmazonS3Config 추가 - AmazonS3 Bean 등록
* FileUploader 에서 AmazonS3 Bean 주입

phs1116 commented on src/main/java/com/wootube/ioi/service/UserService.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    @Transactional
+    public User updateUser(UserSession userSession, EditUserRequestDto editUserRequestDto) {
```
Session에 대한 정보는 Presentaiton Layer에 해당하는 부분므로 도메인 레이어까지 가져가지 않는게 좋을 것 같네요. :)
Session과 상관없는 객체이니 이름을 변경하는 것도 좋을 것 같아요.

MrKwon replied:
UserSession 은 Session 과 관련있는 객체입니다. 😂 추가적으로 메서드 이름이 모호한 것 같아 메서드 이름을 변경하였습니다.

MrKwon replied:
Session 에 대한 정보는 도메인 레이어까지 전달하지 않도록 변경하였습니다 ! 도메인 레이어로 id 를 전달하도록 변경하였습니다 !

phs1116 commented on src/main/java/com/wootube/ioi/service/VideoService.java:
```diff
@@ -0,0 +1,80 @@
+
+@Service
+public class VideoService {
+    private static final String directoryName = "wootube";
```
상수의 컨벤션은 대문자 + underscore 입니다.

그리고 파일 경로가 VideoService의 관심사일까요?

yjll1019 replied:
수정했습니다!
파일 경로는 파일을 저장하는 **FileUploader**에 위치하는게 맞다고 생각되어 해당 클래스로 옮겼습니다!

phs1116 commented on src/main/java/com/wootube/ioi/service/VideoService.java:
```diff
@@ -0,0 +1,80 @@
+
+    private String uploadFile(MultipartFile uploadFile) {
+        try {
+            return fileUploader.uploadFile(uploadFile, directoryName);
```
그저 예외 처리만을 위한 함수인것 같네요. 예외 전환은 정의한 FileUploader에서 할 수 있지 않을까요?

yjll1019 replied:
FileUploader 에서 예외 처리하도록 수정했습니다~!

phs1116 commented on src/main/java/com/wootube/ioi/service/VideoService.java:
```diff
@@ -0,0 +1,80 @@
+        videoRequestDto.setContentPath(videoUrl);
+
+        Video video = modelMapper.map(videoRequestDto, Video.class);
+        video.setOriginFileName(originFileName);
```
ModelMapper가 편하지만 해당 객체가 어디서 생성되는지 추적이 어렵다는 단점도 있습니다.
그리고 setter는 지양하도록 해보세요. :)

또한 다른 곳은 생성자를 통해 생성하는데 컨벤션 또한 서로 안맞는 것 같네요.
컨벤션 통일해주세요. :)

yjll1019 replied:
**1. ModelMapper가 편하지만 해당 객체가 어디서 생성되는지 추적이 어렵다는 단점도 있습니다.**
: 현재 다른 팀원들도 ModelMapper를 사용하고 있어 팀원들과 논의해보겠습니다!

**2. 그리고 setter는 지양하도록 해보세요. :)**
: `video.setOriginFileName(originFileName);` 이 부분에서 사용하고 있는 setter를 제거하기 위한 좋은 방법이 잘 떠오르지 않는데요. 혹시 힌트를 좀 주실 수 있으신가요?ㅠㅠ

**3. 또한 다른 곳은 생성자를 통해 생성하는데 컨벤션 또한 서로 안맞는 것 같네요.
컨벤션 통일해주세요. :)**
: ModelMapper를 이용하여 생성하도록 통일했습니다!

phs1116 replied:
2. modelMapper에 매핑을 하기위한 새로운 변환 객체를 만들던가
그냥 생성자를 통해 생성을 하는 방법이 있을 것 같습니다. :)

phs1116 commented on src/main/java/com/wootube/ioi/service/dto/ReplyRequestDto.java:
```diff
@@ -0,0 +1,12 @@
+import lombok.NoArgsConstructor;
+
+@Getter
+@AllArgsConstructor
```
필드가 적은데 AllArgsConstructor가 필요한가요?
AllArgsConstructor도 편하긴 하지만 필드 추가에 따라 생성자가 어떻게 변경 될지 확인이 되지 않아서
많은 변경이 일어날 때 불편할 수도 있습니다.

phs1116 commented on src/main/java/com/wootube/ioi/service/dto/SignUpRequestDto.java:
```diff
@@ -0,0 +1,16 @@
+import lombok.Getter;
+
+@Getter
+public class SignUpRequestDto {
```
private 기본생성자들도 필요해 보이는데요. 확인 부탁드릴게요. :)

MrKwon replied:
form 으로 받아오는 경우에만 사용되기 때문에  기본 생성자들이 필요없다고 생각하여 기본 생성자를 넣지 않았습니다.
이와 마찬가지로 ``EditUserRequestDto``, ``LogInRequestDto`` 도 기본 생성자를 추가하지 않았는데 추가하는 것이 좋을지 궁금합니다

phs1116 commented on src/main/java/com/wootube/ioi/service/util/FileUploader.java:
```diff
@@ -0,0 +1,64 @@
+import java.util.Optional;
+
+@Component
+public class FileUploader {
```
FileUploader라는것 자체가 여러 케이스로 변경이 일어날 수도 있는 부분으로 보이네요.
인터페이스로 추상화해보면 어떨까요?
그럴 경우 어떤 면에서 장점이 있을까요?

yjll1019 replied:
수정했습니다!
현재 파일을 저장하기 위해 aws만 사용하기 때문에 FileUploader 클래스로 원하는 기능을 수행할 수 있습니다. 하지만 파일을 저장하는 곳이 추가되거나 변경될 경우 새로운 저장소에 데이터를 저장하기 위한 클래스를 생성해야합니다. 
따라서 인터페이스로 추상화한다면 두 클래스 간의 중복 로직을 제거할 수 있습니다. 또,  이를 사용하는 VideoService에서도 코드 변경이 줄어들 수 있습니다.
말씀하신 대로 인터페이스를 사용해서 로직을 분리해보았는데 더 좋은 방법이 있으면 조언 부탁드립니다!

phs1116 commented on src/main/java/com/wootube/ioi/service/util/FileUploader.java:
```diff
@@ -0,0 +1,64 @@
+        this.amazonS3Client = amazonS3Client;
+    }
+
+    @Value("${cloud.aws.s3.bucket}")
```
해당 Value에 대한 선언이 되어있지 않은것 같네요.

yjll1019 replied:
yml 파일에 accessKey와 secretKey 포함되어있어 따로 파일을 업로드하지 않았습니다!
yml파일 추가하고 해당 키는 slack주소 남겨두었습니다!

phs1116 commented on src/main/java/com/wootube/ioi/web/controller/CommentController.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    @DeleteMapping("/{videoId}/comments/{commentId}")
+    public ResponseEntity<Void> deleteComment(@PathVariable Long videoId, @PathVariable Long commentId) {
```
해당 매핑도 Rest API인가요?

phs1116 commented on src/main/java/com/wootube/ioi/web/controller/CommentController.java:
```diff
@@ -0,0 +1,54 @@
+        this.commentService = commentService;
+    }
+
+    @ResponseBody
```
Rest API이면  RestController로 선언해서 사용하시면 ResponseBody는 생략 할 수 있습니다.

phs1116 commented on src/main/java/com/wootube/ioi/web/controller/CommentController.java:
```diff
@@ -0,0 +1,54 @@
+
+import javax.servlet.http.HttpSession;
+
+@RequestMapping("/watch")
```
리소스 명(vedio)과 url이 일치하지 않는것 같네요.
그리고 API라면 Web과의 구분을 통해 api path를 추가해주는게 좋아요!

phs1116 commented on src/main/java/com/wootube/ioi/web/controller/CommentController.java:
```diff
@@ -0,0 +1,54 @@
+        // 로그인 상태인가?
+        // 세션 유저와 댓글 유저와 같은지 확인한다.
+        commentService.delete(commentId);
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
```
ResponseEntity 내부적으로 여러가지 static factory method를 제공합니다.
그것들을 활용해보면 좋을 것 같네요.

hyojaekim replied:
꿀팁 감사합니다 👍

phs1116 commented on src/main/java/com/wootube/ioi/service/UserService.java:
```diff
@@ -0,0 +1,55 @@
+@Service
+public class UserService {
+
+    private UserRepository userRepository;
```
final을 통해서 필수값임을 명시해주는게 좋을 것 같네요.

MrKwon replied:
UserService 뿐만 아니라 다른 부분에도 final 을 통해 필수값을 명시하도록 피드백 반영하였습니다 !

phs1116 commented on src/main/java/com/wootube/ioi/service/VideoService.java:
```diff
@@ -0,0 +1,80 @@
+
+    private final FileUploader fileUploader;
+    private final ModelMapper modelMapper;
+
```
필요없는 공백은 제거해주세요. :)

yjll1019 replied:
네 수정했습니다~!

phs1116 commented on src/main/java/com/wootube/ioi/web/controller/ReplyController.java:
```diff
@@ -0,0 +1,60 @@
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+    }
+
+    @DeleteMapping("/replies/{replyId}")
```
CommentController랑 마찬가지로 참고 부탁드릴게요. :)

phs1116 commented on src/main/java/com/wootube/ioi/service/UserRestService.java:
```diff
@@ -0,0 +1,30 @@
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserRestService {
```
Rest라는 이름 자체는 Domain이 알 필요가 없는 용어입니다. 다른 용어로 변경하는게 좋을 것 같네요. :)

MrKwon replied:
UserSignUpService 로 이름 변경하였습니다 !

phs1116 commented on src/main/java/com/wootube/ioi/service/UserRestService.java:
```diff
@@ -0,0 +1,30 @@
+@Service
+public class UserRestService {
+
+    private static final EmailCheckResponseDto NOT_DUPLICATED_EMAIL_RESPONSE_DTO = new EmailCheckResponseDto("possible");
```
EmailCheckResponseDto 이 해당 용도로만 사용된다면 서비스가 상수로 들고있는것이 아닌
EmailCheckResponseDto 객체 스스로가 가지고있어 static factory method로 제공할 수 있지 않을까요?
또한 응답값이 possible, impossible로 고정되어있다면 enum을 통해 관리도 가능할 것 같습니다.

MrKwon replied:
EmailCheckResponseDto 를 해당 용도로만 사용하기 때문에 static factory method로 제공하도록 변경해 보았습니다.

하지만 enum 을 통해 관리하는 것보다 아직은 한 곳에서만 사용하기 때문에 상수로 빼는게 나을 것 같아 상수로 빼보았습니다.

phs1116 commented on src/test/java/com/wootube/ioi/web/controller/CommentControllerTest.java:
```diff
@@ -0,0 +1,137 @@
+import static org.hamcrest.Matchers.not;
+
+
+public class CommentControllerTest extends CommentCommonControllerTest {
```
ControllerTest가 전체적으로 돌지 않는 것 같은데 확인 부탁드릴게요. :)

phs1116 commented on src/main/java/com/wootube/ioi/domain/validator/PasswordConstraintValidator.java:
```diff
@@ -0,0 +1,43 @@
+
+    public boolean isValid(String password, ConstraintValidatorContext context) {
+        PasswordValidator validator = new PasswordValidator(Arrays.asList(
+                new LengthRule(8, 32),
```
하드코딩보다는 상수로 빼서 의미를 표현해주는게 좋을 것 같네요. :)

MrKwon replied:
피드백 감사합니다 ! 상수로 빼서 의미를 표현하였습니다 !

phs1116 reviewed (changes requested):
안녕하세요! 리뷰어 화투입니다.
서비스 및 컨트롤러간에 컨벤션이 맞지 않은 점
사용하지 않는 주석 코드가 많은점 외에
몇가지 피드백 추가했습니다.
테스트 및 서버 또한 정상적으로 실행되지 않을 것으로 보이는데요 확인 부탁드릴게요. :)
참고 부탁드리고 궁금하신 점 있으시면 연락주세요! :)

SEOKHYOENCHOI committed:
[효오, 비모] ModelMapper 적용 및 Api Contoller 경로 수정 (#40)

* refactor: ModelMapper 적용

* refactor: Comment, Reply Api Controller 경로 및 패키지 수정

backlo committed:
[포도당, 티버] video, index 프론트 (#43)

* chore: build.gradle 초기화

* refactor: index.html 페이지 동적 움직이겠금 수정 및 중복 코드 삭제 및 분할

* chore: build.gradle devtool 추가

* feat: video.html 구현

* feat: livereload 설정

* feat: video.html 코드 수정

* feat: signup.html 과 login.html 작성

* feat: form을 보여주는 컨트롤러 메서드 작성

* signup form
* login form

* chore: WebTestClient 를 위한 Dependency 추가

* test: UserController 회원가입 로그인 폼 메서드

* feat: 회원을 등록하는 기능 추가

* SignUpRequestDto 회원 등록시 필요한 정보를 받아오기 위한 Dto

* test: 회원 추가 기능

* UserServiceTest
* UserControllerTest

* refactor: index페이지로 이동하는 HomeController 생성 및 TestController 삭제

* chore: jupiter, webflux, aws, modelmapper dependency 추가

* feat: 동영상 등록 기능 구현

* chore: aws의 s3 및 파일 업로드 용량 관련 설정 추가

* feat: Video 조회 기능 구현

* feat: Video 수정 기능 구현

* feat: Video 삭제 기능 구현

* chore: static path 경로 지정 삭제

* fix: 리소스 파일 경로 수정

* feat: IndexController 메인화면 보여주는 기능

* feat: findByEmail 추가

* style: SignUpRequestDto 메서드 이름 변경

* test: findByEmail 메서드

* feat: User 패스워드 확인 메서드 구현

* NotMatchPasswordException 추가

* test: matchPassword 메서드 테스트 추가

* feat: 로그인 기능 구현

* LogInRequestDto 추가
* UserController login 메서드
* UserService login 메서드
* NotFoundUserException 예외 클래스

* refactor: video update 메소드 수정

* test: UserService 로그인 성공, 실패 테스트 추가

* test: UserController 유저 로그인 성공, 실패 테스트 추가

* refactor: video 생성 시 해당 페이지로 이동하도록 수정

* refactor: 패키지 구조 변경

* chore: git 리버트

티버 포도당 깃 합치기

* feat: html 과 Controller 연결

* feat: header 로그인 처리 로직 추가

* session에 user 가 있으면 프로필 사진이 나오게 함
* session에 user 가 없으면 로그인 버튼이 나오게 함

* feat: 로그인 세션 처리 기능

* refactor: 패키지 구조 수정

* feat: 로그아웃 기능 추가

* header.html 로그아웃 기능 추가

* refactor: package 구조 수정

* test: test 구조 수정

* feat: video-edit.html 중복 제거

* chore: properties, git ignore 설정

* feat: 마이페이지 추가

* mypage.html
* UserController 마이페이지 메서드

* feat: header 마이페이지, 로그인 버튼 추가

* feat: sidebar.html 홈 href 수정

* chore: build.gradle s3Mock dependency 추가

* refactor: package 구조 변경

* feat: 비디오 프론트 코드 동적으로 수정

* test: video 등록 test 작성

* feat: mypage.html 구현

* feat: User 정보 수정 기능 추가

* EditUserRequestDto : 유저가 입력한 수정 정보를 받아오기 위한 Dto
* User : updateName 메서드 추가
* UserService : update 메서드 추가
* UserController : edit 추가

* feat: mypage.html 회원 탈퇴 버튼 추가

* feat: User 삭제 기능 추가

* UserService delete 메서드 추가, 중복 제거
* UserController deleteUser 메서드 추가, edit 메서드 이름 변경

* test: User 도메인 테스트 접근 제어자 수정

* feat: 아이디 중복 체크 Rest Controller 메서드 추가

* UserRestController
* UserService 중복을 체크하는 checkDuplicate 메서드 추가
* EmailCheckRequestDto 와 EmailCheckRequestDto 추가

* feat: signup.html 중복체크 로직 추가

* chore: Rest Assured 의존성 추가

* feat: EmailCheckResponseDto equals, hashcode 롬복 어노테이션 추가

* test: UserService 아이디 중복 확인 테스트

* test: UserRestController 중복 확인 테스트 추가

* feat: RedirectionArgumentResolver 구현

* 요청을 보낸 주소로 다시 쉽게 Redirect 하기 위해 도와줌

* feat: LoginFailedException 추가

* feat: UserService 로그인 메서드 예외 처리 로직 추가

* 이메일, 비밀번호 둘 중 하나가 틀려도 동일한 메시지를 전달하도록 로직 수정

* test: LoginFailedExceptionAdvice 추가

* feat: login.html 에러를 보여주는 div 태그 추가

* fix: UserService login 예외 처리 로직 변경 관련된 테스트 수정

* feat: 로그인 했을 때를 처리하기 위한 인터셉터

* LoginInterceptor
* LoginInterceptorConfig Config 에 추가

* feat: 로그인 안 했을 때를 처리하기 위한 인터셉터

* NotLoginInterceptor
* NotLoginInterceptorConfig Config 에 추가

* style: image 변경

* test: User updateName 메서드 테스트 추가

* test: IndexController 테스트 추가

* style: UserService 메서드 이름 변경

* refactor: UserRestService 분리

* UserRestController 에서 사용하는 UserService 의 메서드를 UserRestService 로 분리

* test: UserRestService 테스트 작성

* style: signup.html 중복제거 및 코드 분리

* feat: 비디오 삭제 기능 구현

* style: login.html 중복제거 및 코드 분리

* refactor: 로직 단순화

* test: VideoController test code 작성

* style: 클래스 이름 변경

* SessionUser 에서 UserSession 으로 변경
* LoginUserSessionManager 에서 UserSessionManager 로 변경

* refactor: config 클래스명 변경

* refactor: Video 객체의 setContentPath메소드를 사용하는 부분 수정

* fix: 충돌 시 해결하지 않은 부분 수정

* refactor: VideoResponseDto 생성 및 VideoRequestDto 필드 수정

* refactor: Video 클래스 패키지 이동

* style: 코드 컨벤션을 위한 수정

* test: 중복 코드 추출

* style: 상수로 분리

* style: default 생성자 접근 제어자 변경

* EmailCheckRequestDto
* EmailCheckResponseDto

* style: 불필요한 Setter 어노테이션 제거 및 디폴트 생성자 추가

* test: 로그인하지 않고 마이 페이지 요청 시도 테스트 추가

* refactor: Video service 리펙토링

* refactor: Video controller 패키지 리펙토링 및 @AutoConfigureWebTestClient 추가

* test: videos service test 구현

* feat: videos service delete 메소드 deleteById로 변경

* refactor: 코드 리펙토링

* test: test 코드 리펙토링

* refactor: UserSession of 메서드 제거 후 로직 수정

* feat: UserAssembler 추가

* SignUpRequestDto 의 toUser 메서드 제거

* refactor: video.html 리펙토링

* feat: video-edit.html 구조 수정

* feat: video-edit.html upload progress bar 구현

* feat: video-edit.html 구현

* bmodmo

* Merge remote-tracking branch 'luffy/hyobmo' into feat-video-test

# Conflicts:
#	src/main/java/com/wootube/ioi/service/util/FileUploader.java
#	src/main/resources/templates/video.html

* Merge remote-tracking branch 'upstream/develop' into feat-video-test

# Conflicts:
#	src/main/java/com/wootube/ioi/service/util/FileUploader.java
#	src/main/java/com/wootube/ioi/web/support/AmazonS3Config.java
#	src/test/java/com/wootube/ioi/domain/repository/UserRepositoryTest.java
#	src/test/java/com/wootube/ioi/web/controller/CommentCommonControllerTest.java
#	src/test/java/com/wootube/ioi/web/controller/HomeControllerTest.java
#	src/test/java/com/wootube/ioi/web/controller/UserRestControllerTest.java

* feat: index.html 구현

MrKwon committed:
[루피, 티버] 테크코스 S3 연동 (#42)

* feat: signup.html 과 login.html 작성

* feat: form을 보여주는 컨트롤러 메서드 작성

* signup form
* login form

* chore: WebTestClient 를 위한 Dependency 추가

* test: UserController 회원가입 로그인 폼 메서드

* feat: 회원을 등록하는 기능 추가

* SignUpRequestDto 회원 등록시 필요한 정보를 받아오기 위한 Dto

* test: 회원 추가 기능

* UserServiceTest
* UserControllerTest

* chore: static path 경로 지정 삭제

* fix: 리소스 파일 경로 수정

* feat: IndexController 메인화면 보여주는 기능

* feat: findByEmail 추가

* style: SignUpRequestDto 메서드 이름 변경

* test: findByEmail 메서드

* feat: User 패스워드 확인 메서드 구현

* NotMatchPasswordException 추가

* test: matchPassword 메서드 테스트 추가

* feat: 로그인 기능 구현

* LogInRequestDto 추가
* UserController login 메서드
* UserService login 메서드
* NotFoundUserException 예외 클래스

* test: UserService 로그인 성공, 실패 테스트 추가

* test: UserController 유저 로그인 성공, 실패 테스트 추가

* feat: html 과 Controller 연결

* feat: header 로그인 처리 로직 추가

* session에 user 가 있으면 프로필 사진이 나오게 함
* session에 user 가 없으면 로그인 버튼이 나오게 함

* feat: 로그인 세션 처리 기능

* refactor: 패키지 구조 수정

* feat: 로그아웃 기능 추가

* header.html 로그아웃 기능 추가

* feat: 마이페이지 추가

* mypage.html
* UserController 마이페이지 메서드

* feat: header 마이페이지, 로그인 버튼 추가

* feat: sidebar.html 홈 href 수정

* feat: mypage.html 구현

* feat: User 정보 수정 기능 추가

* EditUserRequestDto : 유저가 입력한 수정 정보를 받아오기 위한 Dto
* User : updateName 메서드 추가
* UserService : update 메서드 추가
* UserController : edit 추가

* feat: mypage.html 회원 탈퇴 버튼 추가

* feat: User 삭제 기능 추가

* UserService delete 메서드 추가, 중복 제거
* UserController deleteUser 메서드 추가, edit 메서드 이름 변경

* test: User 도메인 테스트 접근 제어자 수정

* feat: 아이디 중복 체크 Rest Controller 메서드 추가

* UserRestController
* UserService 중복을 체크하는 checkDuplicate 메서드 추가
* EmailCheckRequestDto 와 EmailCheckRequestDto 추가

* feat: signup.html 중복체크 로직 추가

* chore: Rest Assured 의존성 추가

* feat: EmailCheckResponseDto equals, hashcode 롬복 어노테이션 추가

* test: UserService 아이디 중복 확인 테스트

* test: UserRestController 중복 확인 테스트 추가

* feat: RedirectionArgumentResolver 구현

* 요청을 보낸 주소로 다시 쉽게 Redirect 하기 위해 도와줌

* feat: LoginFailedException 추가

* feat: UserService 로그인 메서드 예외 처리 로직 추가

* 이메일, 비밀번호 둘 중 하나가 틀려도 동일한 메시지를 전달하도록 로직 수정

* test: LoginFailedExceptionAdvice 추가

* feat: login.html 에러를 보여주는 div 태그 추가

* fix: UserService login 예외 처리 로직 변경 관련된 테스트 수정

* feat: 로그인 했을 때를 처리하기 위한 인터셉터

* LoginInterceptor
* LoginInterceptorConfig Config 에 추가

* feat: 로그인 안 했을 때를 처리하기 위한 인터셉터

* NotLoginInterceptor
* NotLoginInterceptorConfig Config 에 추가

* style: image 변경

* test: User updateName 메서드 테스트 추가

* test: IndexController 테스트 추가

* style: UserService 메서드 이름 변경

* refactor: UserRestService 분리

* UserRestController 에서 사용하는 UserService 의 메서드를 UserRestService 로 분리

* test: UserRestService 테스트 작성

* style: signup.html 중복제거 및 코드 분리

* style: login.html 중복제거 및 코드 분리

* refactor: 로직 단순화

* style: 클래스 이름 변경

* SessionUser 에서 UserSession 으로 변경
* LoginUserSessionManager 에서 UserSessionManager 로 변경

* style: 상수로 분리

* style: default 생성자 접근 제어자 변경

* EmailCheckRequestDto
* EmailCheckResponseDto

* style: 불필요한 Setter 어노테이션 제거 및 디폴트 생성자 추가

* test: 로그인하지 않고 마이 페이지 요청 시도 테스트 추가

* refactor: UserSession of 메서드 제거 후 로직 수정

* feat: UserAssembler 추가

* SignUpRequestDto 의 toUser 메서드 제거

* chore: spring cloud 관련 dependency 수정

* test: 기존의 테스트 코드로 롤백

* @ExtendWith 어노테이션 삭제

* test: 클래스 이름 변경

* style: 메서드 이름 변경

* fix: aws.yml 관련 설정 제거

* feat: S3 연동 재구현

* AmazonS3Config 추가 - AmazonS3 Bean 등록
* FileUploader 에서 AmazonS3 Bean 주입

* feat: 테크코스의 S3 와 연동

* application.yml 을 .gitignore 에서 제외

* test: 테크코스 S3 연동에 맞게 변경

wmakerjun commented on src/main/resources/static/js/reply.js:
```diff
@@ -0,0 +1,58 @@
+
+        function toggleReplySaveButton(event) {
+            if (event.target.className === "comment-input" && event.target.value !== "") {
+                event.target.parentElement.parentElement.querySelector(".edit").classList.remove("disabled")
```
parentElement를 찾기 위해서 parentElement 메서드를 연속해서 사용하기보다는 [closest](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest)를 이용하는걸 추천합니다. 만약에 parentElment바로 위에 다른 엘리먼트나 요소가 추가됬었을 경우 에러를 발생시킬 수 있기 때문입니다.

wmakerjun commented on src/main/resources/templates/fragments/signup.html:
```diff
@@ -0,0 +1,164 @@
+                passwordCheck: '',
+                rules: {
+                    emailRequired: value => !!value || '이메일을 입력해주세요.',
+                    emailPattern: value => {
```
login.html에서 사용하는 validation 체크와 기능이 유사한데 중복으로 사용되는 같습니다. valdiatio하는 부분을 따로 공통으로 모아서 재사용한다면 더 좋을 것 같아요 :)

MrKwon replied:
피드백 반영하였습니다 !!

wmakerjun commented on src/main/resources/static/js/video.js:
```diff
@@ -0,0 +1,15 @@
+function readMoreTag() {
+    var dots = document.getElementById("dots");
```
var 사용대신에 let과 const를 이용하면 좋을 것 같아요 :)

backlo replied:
반영하였습니다~

wmakerjun reviewed (changes requested):
최신 프레임워크인 Vuejs도 사용하고, ES6 문법을 적극적으로 사용한점이 너무 멋집니다 💯 
프론트엔드에서 변수사용하는 것과, 약간의 중복만 제거하면 더 멋진 코드가 될 것 같습니다. 😃

woowahan-pjs reviewed (approved):
아직 피드백이 반영되지 않았어요!
피드백 반영 후 개선할 점들을 더 찾아봐야겠어요.
발표 전까지 힘내 주길 바래요.

backlo committed:
[포도당] Video 코드의 피드백 반영 (#48)

* chore: build.gradle 초기화

* refactor: index.html 페이지 동적 움직이겠금 수정 및 중복 코드 삭제 및 분할

* chore: build.gradle devtool 추가

* feat: video.html 구현

* feat: livereload 설정

* feat: video.html 코드 수정

* feat: signup.html 과 login.html 작성

* feat: form을 보여주는 컨트롤러 메서드 작성

* signup form
* login form

* chore: WebTestClient 를 위한 Dependency 추가

* test: UserController 회원가입 로그인 폼 메서드

* feat: 회원을 등록하는 기능 추가

* SignUpRequestDto 회원 등록시 필요한 정보를 받아오기 위한 Dto

* test: 회원 추가 기능

* UserServiceTest
* UserControllerTest

* refactor: index페이지로 이동하는 HomeController 생성 및 TestController 삭제

* chore: jupiter, webflux, aws, modelmapper dependency 추가

* feat: 동영상 등록 기능 구현

* chore: aws의 s3 및 파일 업로드 용량 관련 설정 추가

* feat: Video 조회 기능 구현

* feat: Video 수정 기능 구현

* feat: Video 삭제 기능 구현

* chore: static path 경로 지정 삭제

* fix: 리소스 파일 경로 수정

* feat: IndexController 메인화면 보여주는 기능

* feat: findByEmail 추가

* style: SignUpRequestDto 메서드 이름 변경

* test: findByEmail 메서드

* feat: User 패스워드 확인 메서드 구현

* NotMatchPasswordException 추가

* test: matchPassword 메서드 테스트 추가

* feat: 로그인 기능 구현

* LogInRequestDto 추가
* UserController login 메서드
* UserService login 메서드
* NotFoundUserException 예외 클래스

* refactor: video update 메소드 수정

* test: UserService 로그인 성공, 실패 테스트 추가

* test: UserController 유저 로그인 성공, 실패 테스트 추가

* refactor: video 생성 시 해당 페이지로 이동하도록 수정

* refactor: 패키지 구조 변경

* chore: git 리버트

티버 포도당 깃 합치기

* feat: html 과 Controller 연결

* feat: header 로그인 처리 로직 추가

* session에 user 가 있으면 프로필 사진이 나오게 함
* session에 user 가 없으면 로그인 버튼이 나오게 함

* feat: 로그인 세션 처리 기능

* refactor: 패키지 구조 수정

* feat: 로그아웃 기능 추가

* header.html 로그아웃 기능 추가

* refactor: package 구조 수정

* test: test 구조 수정

* feat: video-edit.html 중복 제거

* chore: properties, git ignore 설정

* feat: 마이페이지 추가

* mypage.html
* UserController 마이페이지 메서드

* feat: header 마이페이지, 로그인 버튼 추가

* feat: sidebar.html 홈 href 수정

* chore: build.gradle s3Mock dependency 추가

* refactor: package 구조 변경

* feat: 비디오 프론트 코드 동적으로 수정

* test: video 등록 test 작성

* feat: mypage.html 구현

* feat: User 정보 수정 기능 추가

* EditUserRequestDto : 유저가 입력한 수정 정보를 받아오기 위한 Dto
* User : updateName 메서드 추가
* UserService : update 메서드 추가
* UserController : edit 추가

* feat: mypage.html 회원 탈퇴 버튼 추가

* feat: User 삭제 기능 추가

* UserService delete 메서드 추가, 중복 제거
* UserController deleteUser 메서드 추가, edit 메서드 이름 변경

* test: User 도메인 테스트 접근 제어자 수정

* feat: 아이디 중복 체크 Rest Controller 메서드 추가

* UserRestController
* UserService 중복을 체크하는 checkDuplicate 메서드 추가
* EmailCheckRequestDto 와 EmailCheckRequestDto 추가

* feat: signup.html 중복체크 로직 추가

* chore: Rest Assured 의존성 추가

* feat: EmailCheckResponseDto equals, hashcode 롬복 어노테이션 추가

* test: UserService 아이디 중복 확인 테스트

* test: UserRestController 중복 확인 테스트 추가

* feat: RedirectionArgumentResolver 구현

* 요청을 보낸 주소로 다시 쉽게 Redirect 하기 위해 도와줌

* feat: LoginFailedException 추가

* feat: UserService 로그인 메서드 예외 처리 로직 추가

* 이메일, 비밀번호 둘 중 하나가 틀려도 동일한 메시지를 전달하도록 로직 수정

* test: LoginFailedExceptionAdvice 추가

* feat: login.html 에러를 보여주는 div 태그 추가

* fix: UserService login 예외 처리 로직 변경 관련된 테스트 수정

* feat: 로그인 했을 때를 처리하기 위한 인터셉터

* LoginInterceptor
* LoginInterceptorConfig Config 에 추가

* feat: 로그인 안 했을 때를 처리하기 위한 인터셉터

* NotLoginInterceptor
* NotLoginInterceptorConfig Config 에 추가

* style: image 변경

* test: User updateName 메서드 테스트 추가

* test: IndexController 테스트 추가

* style: UserService 메서드 이름 변경

* refactor: UserRestService 분리

* UserRestController 에서 사용하는 UserService 의 메서드를 UserRestService 로 분리

* test: UserRestService 테스트 작성

* style: signup.html 중복제거 및 코드 분리

* feat: 비디오 삭제 기능 구현

* style: login.html 중복제거 및 코드 분리

* refactor: 로직 단순화

* test: VideoController test code 작성

* style: 클래스 이름 변경

* SessionUser 에서 UserSession 으로 변경
* LoginUserSessionManager 에서 UserSessionManager 로 변경

* refactor: config 클래스명 변경

* refactor: Video 객체의 setContentPath메소드를 사용하는 부분 수정

* fix: 충돌 시 해결하지 않은 부분 수정

* refactor: VideoResponseDto 생성 및 VideoRequestDto 필드 수정

* refactor: Video 클래스 패키지 이동

* style: 코드 컨벤션을 위한 수정

* test: 중복 코드 추출

* style: 상수로 분리

* style: default 생성자 접근 제어자 변경

* EmailCheckRequestDto
* EmailCheckResponseDto

* style: 불필요한 Setter 어노테이션 제거 및 디폴트 생성자 추가

* test: 로그인하지 않고 마이 페이지 요청 시도 테스트 추가

* refactor: Video service 리펙토링

* refactor: Video controller 패키지 리펙토링 및 @AutoConfigureWebTestClient 추가

* test: videos service test 구현

* feat: videos service delete 메소드 deleteById로 변경

* refactor: 코드 리펙토링

* test: test 코드 리펙토링

* refactor: UserSession of 메서드 제거 후 로직 수정

* feat: UserAssembler 추가

* SignUpRequestDto 의 toUser 메서드 제거

* refactor: video.html 리펙토링

* feat: video-edit.html 구조 수정

* feat: video-edit.html upload progress bar 구현

* feat: video-edit.html 구현

* bmodmo

* Merge remote-tracking branch 'luffy/hyobmo' into feat-video-test

# Conflicts:
#	src/main/java/com/wootube/ioi/service/util/FileUploader.java
#	src/main/resources/templates/video.html

* Merge remote-tracking branch 'upstream/develop' into feat-video-test

# Conflicts:
#	src/main/java/com/wootube/ioi/service/util/FileUploader.java
#	src/main/java/com/wootube/ioi/web/support/AmazonS3Config.java
#	src/test/java/com/wootube/ioi/domain/repository/UserRepositoryTest.java
#	src/test/java/com/wootube/ioi/web/controller/CommentCommonControllerTest.java
#	src/test/java/com/wootube/ioi/web/controller/HomeControllerTest.java
#	src/test/java/com/wootube/ioi/web/controller/UserRestControllerTest.java

* feat: index.html 구현

* video.js 피드백 반영

* Merge remote-tracking branch 'upstream/develop' into feat-video-test

# Conflicts:
#	src/main/java/com/wootube/ioi/web/support/AmazonS3Config.java
#	src/test/java/com/wootube/ioi/web/config/TestConfig.java

hyojaekim committed:
[효오, 비모] Comment 코드의 피드백 반영 (#50)

* style: 불필요한 주석 제거

* 피드백 반영

* style: 불필요한 주석 및 사용하지 않는 코드 제거

* refactor: Access Level 추가

* refactor: 불필요한 HTML 코드 제거

* refactor: Rest Controller 어노테이션 추가

* refactor: 공통된 필드 중복 제거

* refactor: S3 accessKey, secretKey 필드 추가

* refactor: ResponseEntity factory method 적용

MrKwon committed:
[루피] User 코드, 프론트코드의 피드백을 반영한다. (#46)

* feat: signup.html 과 login.html 작성

* feat: form을 보여주는 컨트롤러 메서드 작성

* signup form
* login form

* chore: WebTestClient 를 위한 Dependency 추가

* test: UserController 회원가입 로그인 폼 메서드

* feat: 회원을 등록하는 기능 추가

* SignUpRequestDto 회원 등록시 필요한 정보를 받아오기 위한 Dto

* test: 회원 추가 기능

* UserServiceTest
* UserControllerTest

* chore: static path 경로 지정 삭제

* fix: 리소스 파일 경로 수정

* feat: IndexController 메인화면 보여주는 기능

* feat: findByEmail 추가

* style: SignUpRequestDto 메서드 이름 변경

* test: findByEmail 메서드

* feat: User 패스워드 확인 메서드 구현

* NotMatchPasswordException 추가

* test: matchPassword 메서드 테스트 추가

* feat: 로그인 기능 구현

* LogInRequestDto 추가
* UserController login 메서드
* UserService login 메서드
* NotFoundUserException 예외 클래스

* test: UserService 로그인 성공, 실패 테스트 추가

* test: UserController 유저 로그인 성공, 실패 테스트 추가

* feat: html 과 Controller 연결

* feat: header 로그인 처리 로직 추가

* session에 user 가 있으면 프로필 사진이 나오게 함
* session에 user 가 없으면 로그인 버튼이 나오게 함

* feat: 로그인 세션 처리 기능

* refactor: 패키지 구조 수정

* feat: 로그아웃 기능 추가

* header.html 로그아웃 기능 추가

* feat: 마이페이지 추가

* mypage.html
* UserController 마이페이지 메서드

* feat: header 마이페이지, 로그인 버튼 추가

* feat: sidebar.html 홈 href 수정

* feat: mypage.html 구현

* feat: User 정보 수정 기능 추가

* EditUserRequestDto : 유저가 입력한 수정 정보를 받아오기 위한 Dto
* User : updateName 메서드 추가
* UserService : update 메서드 추가
* UserController : edit 추가

* feat: mypage.html 회원 탈퇴 버튼 추가

* feat: User 삭제 기능 추가

* UserService delete 메서드 추가, 중복 제거
* UserController deleteUser 메서드 추가, edit 메서드 이름 변경

* test: User 도메인 테스트 접근 제어자 수정

* feat: 아이디 중복 체크 Rest Controller 메서드 추가

* UserRestController
* UserService 중복을 체크하는 checkDuplicate 메서드 추가
* EmailCheckRequestDto 와 EmailCheckRequestDto 추가

* feat: signup.html 중복체크 로직 추가

* chore: Rest Assured 의존성 추가

* feat: EmailCheckResponseDto equals, hashcode 롬복 어노테이션 추가

* test: UserService 아이디 중복 확인 테스트

* test: UserRestController 중복 확인 테스트 추가

* feat: RedirectionArgumentResolver 구현

* 요청을 보낸 주소로 다시 쉽게 Redirect 하기 위해 도와줌

* feat: LoginFailedException 추가

* feat: UserService 로그인 메서드 예외 처리 로직 추가

* 이메일, 비밀번호 둘 중 하나가 틀려도 동일한 메시지를 전달하도록 로직 수정

* test: LoginFailedExceptionAdvice 추가

* feat: login.html 에러를 보여주는 div 태그 추가

* fix: UserService login 예외 처리 로직 변경 관련된 테스트 수정

* feat: 로그인 했을 때를 처리하기 위한 인터셉터

* LoginInterceptor
* LoginInterceptorConfig Config 에 추가

* feat: 로그인 안 했을 때를 처리하기 위한 인터셉터

* NotLoginInterceptor
* NotLoginInterceptorConfig Config 에 추가

* style: image 변경

* test: User updateName 메서드 테스트 추가

* test: IndexController 테스트 추가

* style: UserService 메서드 이름 변경

* refactor: UserRestService 분리

* UserRestController 에서 사용하는 UserService 의 메서드를 UserRestService 로 분리

* test: UserRestService 테스트 작성

* style: signup.html 중복제거 및 코드 분리

* style: login.html 중복제거 및 코드 분리

* refactor: 로직 단순화

* style: 클래스 이름 변경

* SessionUser 에서 UserSession 으로 변경
* LoginUserSessionManager 에서 UserSessionManager 로 변경

* style: 상수로 분리

* style: default 생성자 접근 제어자 변경

* EmailCheckRequestDto
* EmailCheckResponseDto

* style: 불필요한 Setter 어노테이션 제거 및 디폴트 생성자 추가

* test: 로그인하지 않고 마이 페이지 요청 시도 테스트 추가

* refactor: UserSession of 메서드 제거 후 로직 수정

* feat: UserAssembler 추가

* SignUpRequestDto 의 toUser 메서드 제거

* chore: spring cloud 관련 dependency 수정

* test: 기존의 테스트 코드로 롤백

* @ExtendWith 어노테이션 삭제

* test: 클래스 이름 변경

* style: 메서드 이름 변경

* fix: aws.yml 관련 설정 제거

* feat: S3 연동 재구현

* AmazonS3Config 추가 - AmazonS3 Bean 등록
* FileUploader 에서 AmazonS3 Bean 주입

* feat: 테크코스의 S3 와 연동

* application.yml 을 .gitignore 에서 제외

* test: 테크코스 S3 연동에 맞게 변경

* feat: User EqualsAndHashCode 변경

* refactor: UserService, Assembler 를 ModelMapper 로 교체

* UserAssembler 삭제
* UserService 에 ModelMapper 추가

* test: UserServiceTest ModelMapper Mock 추가

* test: UserRepositoryTest 삭제

* refactor: UserController 에서 전달하는 정보 수정

* UserSessionManager - 메서드 이름 변경
* UserService - updateUser, createUser 메서드에 전달 받는 정보를 userSession 에서 id 로 변경
* UserController - UserService 변경에 따른 변경

* style: final 추가하여 필수값 명시

* style: 클래스 이름 변경

* refactor: UserSignUpService 상수 제거

* EmailCheckResponseDto 에 impossible, possible 메서드 추가

* style: PasswordConstraintValidator 상수로 분리

* fix: UserSignUpService 응답 수정

* refactor: mypage.html vue 를 이용한 이벤트 핸들링으로 변경

* refactor: signup.html vue 를 이용한 이벤트 핸들링으로 변경

* refactor: html 중복제거

yjll1019 committed:
[티버] Video 코드의  피드백 반영 (#49)

* chore: build.gradle 초기화

* refactor: index.html 페이지 동적 움직이겠금 수정 및 중복 코드 삭제 및 분할

* chore: build.gradle devtool 추가

* feat: video.html 구현

* feat: livereload 설정

* feat: video.html 코드 수정

* refactor: index페이지로 이동하는 HomeController 생성 및 TestController 삭제

* chore: jupiter, webflux, aws, modelmapper dependency 추가

* feat: 동영상 등록 기능 구현

* chore: aws의 s3 및 파일 업로드 용량 관련 설정 추가

* feat: Video 조회 기능 구현

* feat: Video 수정 기능 구현

* feat: Video 삭제 기능 구현

* refactor: video update 메소드 수정

* refactor: video 생성 시 해당 페이지로 이동하도록 수정

* refactor: 패키지 구조 변경

* chore: git 리버트

티버 포도당 깃 합치기

* refactor: package 구조 수정

* test: test 구조 수정

* feat: video-edit.html 중복 제거

* chore: properties, git ignore 설정

* chore: build.gradle s3Mock dependency 추가

* refactor: package 구조 변경

* feat: 비디오 프론트 코드 동적으로 수정

* test: video 등록 test 작성

* feat: 비디오 삭제 기능 구현

* test: VideoController test code 작성

* refactor: config 클래스명 변경

* refactor: Video 객체의 setContentPath메소드를 사용하는 부분 수정

* fix: 충돌 시 해결하지 않은 부분 수정

* refactor: VideoResponseDto 생성 및 VideoRequestDto 필드 수정

* refactor: Video 클래스 패키지 이동

* style: 코드 컨벤션을 위한 수정

* test: 중복 코드 추출

* refactor: Video service 리펙토링

* refactor: Video controller 패키지 리펙토링 및 @AutoConfigureWebTestClient 추가

* test: videos service test 구현

* feat: videos service delete 메소드 deleteById로 변경

* refactor: 코드 리펙토링

* test: test 코드 리펙토링

* Merge remote-tracking branch 'upstream/develop'

# Conflicts:
#	build.gradle
#	src/main/java/com/wootube/ioi/WootubeApplication.java
#	src/main/java/com/wootube/ioi/domain/model/User.java
#	src/main/java/com/wootube/ioi/service/util/FileUploader.java
#	src/main/java/com/wootube/ioi/web/controller/HomeController.java
#	src/main/java/com/wootube/ioi/web/support/ModelConfig.java
#	src/main/resources/application.properties
#	src/main/resources/static/css/app.css
#	src/main/resources/static/css/video-edit.css
#	src/main/resources/templates/fragments/header.html
#	src/main/resources/templates/index.html
#	src/main/resources/templates/video-edit.html
#	src/main/resources/templates/video.html
#	src/test/java/com/wootube/ioi/domain/model/UserTest.java
#	src/test/java/com/wootube/ioi/web/config/TestConfig.java
#	src/test/java/com/wootube/ioi/web/controller/VideoControllerTest.java
#	src/test/resources/application.properties

* fix: aws s3 연동 방법 수정

* refactor: Video의 lombok 어노테이션 수정

- Setter 제거
- EqualsAndHashCode 추가
- NoArgsConstructor AccessLevel 설정

* test: Video 클래스 lombok 어노테이션 수정 후 테스트 코드 변경

* refactor: 파일 저장 위치를 나타내는 변수 FileUploader로 이동

* test: uploadFile메소드 호출 시 파일 경로를 나타내는 매개변수 제거

* refactor: 설정 파일 관련 코드 제거

* refactor: 중복 필드 제거

* refactor: FileUploader 를 인터페이스로 추상화

yjll1019 committed:
[티버] Video 코드의 피드백 반영 (#57)

* chore: build.gradle 초기화

* refactor: index.html 페이지 동적 움직이겠금 수정 및 중복 코드 삭제 및 분할

* chore: build.gradle devtool 추가

* feat: video.html 구현

* feat: livereload 설정

* feat: video.html 코드 수정

* refactor: index페이지로 이동하는 HomeController 생성 및 TestController 삭제

* chore: jupiter, webflux, aws, modelmapper dependency 추가

* feat: 동영상 등록 기능 구현

* chore: aws의 s3 및 파일 업로드 용량 관련 설정 추가

* feat: Video 조회 기능 구현

* feat: Video 수정 기능 구현

* feat: Video 삭제 기능 구현

* refactor: video update 메소드 수정

* refactor: video 생성 시 해당 페이지로 이동하도록 수정

* refactor: 패키지 구조 변경

* chore: git 리버트

티버 포도당 깃 합치기

* refactor: package 구조 수정

* test: test 구조 수정

* feat: video-edit.html 중복 제거

* chore: properties, git ignore 설정

* chore: build.gradle s3Mock dependency 추가

* refactor: package 구조 변경

* feat: 비디오 프론트 코드 동적으로 수정

* test: video 등록 test 작성

* feat: 비디오 삭제 기능 구현

* test: VideoController test code 작성

* refactor: config 클래스명 변경

* refactor: Video 객체의 setContentPath메소드를 사용하는 부분 수정

* fix: 충돌 시 해결하지 않은 부분 수정

* refactor: VideoResponseDto 생성 및 VideoRequestDto 필드 수정

* refactor: Video 클래스 패키지 이동

* style: 코드 컨벤션을 위한 수정

* test: 중복 코드 추출

* refactor: Video service 리펙토링

* refactor: Video controller 패키지 리펙토링 및 @AutoConfigureWebTestClient 추가

* test: videos service test 구현

* feat: videos service delete 메소드 deleteById로 변경

* refactor: 코드 리펙토링

* test: test 코드 리펙토링

* Merge remote-tracking branch 'upstream/develop'

# Conflicts:
#	build.gradle
#	src/main/java/com/wootube/ioi/WootubeApplication.java
#	src/main/java/com/wootube/ioi/domain/model/User.java
#	src/main/java/com/wootube/ioi/service/util/FileUploader.java
#	src/main/java/com/wootube/ioi/web/controller/HomeController.java
#	src/main/java/com/wootube/ioi/web/support/ModelConfig.java
#	src/main/resources/application.properties
#	src/main/resources/static/css/app.css
#	src/main/resources/static/css/video-edit.css
#	src/main/resources/templates/fragments/header.html
#	src/main/resources/templates/index.html
#	src/main/resources/templates/video-edit.html
#	src/main/resources/templates/video.html
#	src/test/java/com/wootube/ioi/domain/model/UserTest.java
#	src/test/java/com/wootube/ioi/web/config/TestConfig.java
#	src/test/java/com/wootube/ioi/web/controller/VideoControllerTest.java
#	src/test/resources/application.properties

* fix: aws s3 연동 방법 수정

* refactor: Video의 lombok 어노테이션 수정

- Setter 제거
- EqualsAndHashCode 추가
- NoArgsConstructor AccessLevel 설정

* test: Video 클래스 lombok 어노테이션 수정 후 테스트 코드 변경

* refactor: 파일 저장 위치를 나타내는 변수 FileUploader로 이동

* test: uploadFile메소드 호출 시 파일 경로를 나타내는 매개변수 제거

* refactor: 설정 파일 관련 코드 제거

* refactor: 중복 필드 제거

* refactor: FileUploader 를 인터페이스로 추상화

* chore: application.yml 수정
